tests/lib/bin/TODO

```diff
@@ -0,0 +1,115 @@
+ONCE THINGS ARE MADE UNIFORM THIS SHOULD BECOME A README WITHOUT THE TODO PART!
+
+This "tools" directory collects useful utility commands that can be
+used in our spread test suites. (They are separate from the shell
+function libraries *.sh in the level above.)
+
+Commands here should have names that clarify their purpose, longish
+names using '-' as word separator are expected. Commands that take
+themselves a full command to execute should usually end in '-do'.
+
+The commands might take one or multiple subcommands themselves if
+meaningful, subcommands should also use '-' as internal word
+separator.
+
+If a subcommand executes other commands it should usually be called
+"exec".
+
+All the commands invoked without arguments should either do something
+informative/non-destructive or print usage.
+
+All the commands should support at least a -h help option.
+
+To invoke a command here tests can do "$TESTSTOOLS"/cmd.
+
+Commands that are used very often or that are particularly useful to
+debug at a spread -debug shell prompt can be promoted to be on the
+PATH during tests, if an acceptable name can be found for them.
+
+This is done by putting symlinks to here from tests/bin which itself
+will be on the PATH.
+
+Commands expected to be on the PATH should follow more specific naming
+conventions, the name should either hint at testing or trade that off
+for brevity, but only when appropriate:
+
+* really frequently used commands can have short/pithy names (not, retry...)
+
+* commands that manage or deal with cross-test/cross-task state should be
+  prefixed with "tests." (tests.session)
+
+* otherwise the command should take some form of suffix like ".query", ".check"
+  that hints at observing, checking the system
+
+In general if none of these apply they should continue to be invoked
+via $TESTTOOLS, or possibly have a tests/lib/*.sh wrapping them if
+appropriate.
+
+TODO:
+
+rename tests/lib/bin to tests/lib/tools, define TESTSTOOLS in
+spread.yaml pointing at it
+
+create tests/bin, have ultimately this on the PATH and not tools
+
+symlinks and renames (plus fixes):
+
+tests/bin/MATCH -> tests/lib/tools/MATCH
+tests/bin/REBOOT -> tests/lib/tools/REBOOT
+tests/bin/not -> tests/lib/tools/not
+tests/bin/retry -> tests/lib/tools/retry (was retry-tool)
+
+tests/bin/tests.session -> tests/lib/tools/tests.session (was session-tool)
+  fix to have -h
```

Comment:
Those are done, exec is not required because of PRs in flight, but can be easily made so by removing the fallback case.